connectors/src/resources/slack_configuration_resource.ts

```diff
@@ -0,0 +1,120 @@
+import type { ModelId, Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type { Attributes, ModelStatic, Transaction } from "sequelize";
+
+import { SlackConfigurationModel } from "@connectors/lib/models/slack";
+import { BaseResource } from "@connectors/resources/base_resource";
+import { sequelizeConnection } from "@connectors/resources/storage";
+import type { ReadonlyAttributesType } from "@connectors/resources/storage/types";
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SlackConfigurationResource
+  extends ReadonlyAttributesType<SlackConfigurationModel> {}
+export class SlackConfigurationResource extends BaseResource<SlackConfigurationModel> {
+  static model: ModelStatic<SlackConfigurationModel> = SlackConfigurationModel;
+
+  constructor(
+    model: ModelStatic<SlackConfigurationModel>,
+    blob: Attributes<SlackConfigurationModel>
+  ) {
+    super(SlackConfigurationModel, blob);
+  }
+
+  static async makeNew({
+    slackTeamId,
+    connectorId,
+    transaction,
+  }: {
+    slackTeamId: string;
+    connectorId: ModelId;
+    transaction: Transaction;
+  }) {
+    const otherSlackConfigurationWithBotEnabled =
+      await SlackConfigurationModel.findOne({
+        where: {
+          slackTeamId,
+          botEnabled: true,
+        },
+        transaction,
+      });
+
+    await SlackConfigurationModel.create(
+      {
+        slackTeamId,
+        botEnabled: otherSlackConfigurationWithBotEnabled ? false : true,
+        connectorId: connectorId,
+      },
+      { transaction }
+    );
+  }
+
+  static async fetchByConnectorId(id: ModelId) {
```

Comment:
`nit` We could even go one step further and takes a `ConnectorResource` to strengthen the relationship between `ConnectorResource` and `SlackConfigurationResource`.

```suggestion
  static async fetchByConnectorId(connectorId: ModelId) {
```